build/android/cross

```diff
@@ -3,36 +3,24 @@ set -e
 
 ROOTDIR=$(cd $(dirname $0) && pwd -P)
 
-if [ $# -lt 2 ]; then
-    echo "Usage: $0 ARCH API" 1>&2
-    echo "  ARCH: aarch64-linux-android arm-linux-androideabi mipsel-linux-android mips64el-linux-android x86 x86_64" 1>&2
-    echo "  API: 1 ... 21" 1>&2
+if [ $# -lt 1 ]; then
+    echo "Usage: $0 ARCH" 1>&2
+    echo "" 1>&2
+    echo "  ARCH: armeabi, armeabi-v7a, arm64, mips, mips64, x86, x86_64" 1>&2
+    echo "" 1>&2
     echo "Example usage:" 1>&2
-    echo " $0 aarch64-linux-android 21" 1>&2
+    echo " $0 arm64" 1>&2
     exit 1
 fi
 
 ARCH=$1
 shift 
-API=$1
-shift
 
 echo ""
-echo "Cross compiling for $ARCH with api: $API"
-
-# XXX shortcut for armeabi-v7a
-# According https://developer.android.com/ndk/guides/standalone_toolchain.html
-# the proper solution is to override CFLAGS
-if [ $ARCH = "arm-linux-androideabi-v7a" ]; then
-    echo "WARNING: converting $ARCH to arm-linux-androideabi" 1>&2
-    ARCH="arm-linux-androideabi"
-    V7A="-v7a"
-else
-    V7A=""
-fi
```

Comment:
This diff also improves the way in which we compile for arm. We now pass the same flags as ndk-build and we deal properly with both armv5 and armv7.